src/vanilla/Model/PrimaryTypeTS.cs

```diff
@@ -58,7 +58,10 @@ public virtual string ImplementationName
                         return "Object";
 
                     case KnownPrimaryType.Credentials:
-                        return "msRest.ServiceClientCredentials";
+                        return "msRest.ServiceClientCredentials" + 
+                            ((CodeModel as CodeModelTS).Settings.UseAzureCore ?? false
+                                ? " | msRest.TokenCredential"
+                                : string.Empty);

```

Comment:
I noticed this code was duplicated from above. Please extract into a function so that the logic stays in one place.